### ISSUE TITLE
Won't build on ARM / ubuntu 14.04

### ISSUE THREAD

**Issue by techhazard** [atom/atom#3783]
While building the editor according to the instructions for ubuntu 14.04 on armhf, I get this error:

```
Node: v0.10.32
npm: v1.4.28
Installing build modules...
Installing apm...
npm WARN package.json github-url-from-git@1.1.1 No repository field.

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: incorrect header check
    at Zlib._binding.onerror (zlib.js:295:17)

npm ERR! atom-package-manager@0.101.0 install: `node ./script/download-node.js`
npm ERR! Exit status 8
npm ERR!
npm ERR! Failed at the atom-package-manager@0.101.0 install script.
npm ERR! This is most likely a problem with the atom-package-manager package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./script/download-node.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls atom-package-manager
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.0.31-1759357
npm ERR! command "node" "/root/git/atom2/atom/build/node_modules/.bin/npm" "--userconfig=/root/git/atom2/atom/.npmrc" "install" "--quiet"
npm ERR! cwd /root/git/atom2/atom/apm
npm ERR! node -v v0.10.32
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
```

I am running all commands as root.'What am I missing?

**Comment by henrywright** [comment]
I need this too. Wished I'd bought an Intel Chromebook, these ARM processors are troublesome :(

**Comment by pwaring** [comment]
Would the contents of `apm/npm-debug.log` assist with fixing this bug? I was going to post it before realising that it was 1.3Mb in size.

**Comment by jbt** [comment]
[hmm I swear I commented this just now, before opening that issue on apm but it appears to have vanished. Maybe I never hit submit]

As I mentioned in atom/apm#312 this looks to be happening because apm is trying to grab node v0.10.35, which doesn't distribute a prebuilt armv7l binary, and it's trying to grab http://nodejs.org/dist/v0.10.35/node-v0.10.35-linux-armv7l.tar.gz which is a 404 and is confusing zlib.

I'm guessing the only options would be to switch to iojs (which does include armv6l/armv7l binaries, but could break many other things which I don't know about) or somehow get apm to download and build from source if it can't grab the binary (which would be horrible and messy obviously)

**Comment by thedaniel** [comment]
>  the only options would be to switch to iojs

Atom-shell recently switched to io.js, and so did Atom as of https://github.com/atom/atom/releases/tag/v0.177.0

**Comment by jbt** [comment]
Not in https://github.com/atom/apm/blob/master/script/download-node.js it didn't, which is what's causing this error.

**Comment by thedaniel** [comment]
Ah, the node bundled with apm, of course.

**Comment by johnjelinek** [comment]
Any updates on upgrading to iojs (and thus supporting arm) in `download-node.js`?

**Comment by thedaniel** [comment]
@johnjelinek we already use iojs, supporting arm is a lot more than that.

**Comment by JohnMurga** [comment]
The initial Atom build errors are possible to work around ...
On my RPi2 I built Node v0.10.35 separately and supplanted `download-node.js`.

The problem is that later it'll try and download another blob for Electron, which currently doesn't build for ARM either (although there are versions of NW.JS that do).

So it's possible to get to this point, before you are well and truly stuck :-(
```
Deduping modules √
Running "download-atom-shell" task
>> Cannot find atom-shell-v0.22.3-linux-arm.zip in electron v0.22.3 release
Aborted due to warnings.
```

**Comment by thedaniel** [comment]
https://github.com/atom/electron/issues/366#issuecomment-113033739 if you'd really like to get cracking on this you can try to get Electron to build on ARM - and per the issue linked above to do that you have to get libchromiumcontent to build on ARM. @jlord has been doing some work in this area recently as well.

**Comment by jlord** [comment]
:wave: I've been trying to get Atom up and running on ARM and created an issue with updates linked :point_up: 

tl;dr it builds and launches but lots of errors.

**Comment by scls19fr** [comment]
+1 having Atom on my Raspberry Pi 2 will be great!

**Comment by lee-dohm** [comment]
We don't support Atom on the Raspberry Pi. While we would love to support Atom on all possible platforms, our resources are finite so we have to draw a line somewhere. You can find out more information about Atom on ARM and links to other community efforts on #7822.

**Comment by curiousercreative** [comment]
It's been announced on #7822, but just for those stumbling upon this, we have a community fork for Atom on ARM that is seemingly stable to build and install. If you find otherwise, please open an issue on the forked repo: https://github.com/hypersad/atom
